src/anomalib/models/image/uflow/nfa_tree.py

```diff
@@ -0,0 +1,167 @@
+import itertools as it
+
+import networkx as nx
+import numpy as np
+import torch
+from mpmath import mp
+from skimage.morphology import max_tree
+
+mp.dps = 15
+
+
+class NFATree:
```

Comment:
Can you add typing and docstring to the methods of this class?

Reply:
Also added. Thanks!